packages/store/ts/codegen/field.ts

```diff
@@ -137,6 +140,15 @@ export function renderFieldMethods(options: RenderTableOptions): string {
                 "uint256 _index",
               ])}) internal view returns (${portionData.typeWithLocation}) {
                 ${_keyTupleDefinition}
+
+                uint256 _byteLength = ${_store}.getDynamicFieldLength(_tableId, _keyTuple, ${dynamicSchemaIndex});
+                uint256 dynamicLength = _byteLength / ${portionData.elementLength};
+                uint256 staticLength = ${typeWrappingData.staticLength};
+
+                if (_index < staticLength && _index >= dynamicLength) {
+                  return 0;
+                }
+
```

Comment:
could we just have the `if (typeWrappingData && typeWrappingData.kind === "staticArray")` check around this small block?

another question, is `0` implicitly convertible to the expected return type or do we need to do some decoding/casting here?

Reply:
1) Good point, we don't need this big diff

2) We need casting, I'm still figuring out which helpers to use 💪

Reply:
Done both these!